### PR TITLE
Ensure deployments spread over availability zones

### PR DESCRIPTION
**Pull request by redhug1**
### What

As per title
(i've already tested a patched version of the nomad file in Sandbox for publishing and babbage in pubishing has now been spread correctly, where before they were not)

### How to review

Do changes look correct

### Who can review

Platform, Gedge, James A.

### CODE REVIEW

File: babbage.nomad [hunk 1]
```diff
@@ -101,8 +106,15 @@ job "babbage" {
   group "publishing" {
     count = "{{PUBLISHING_TASK_COUNT}}"
 
-    constraint {
-      distinct_hosts = true
+    spread {
+      attribute = "${node.unique.id}"
+      weight    = 100
```
Comment (gedge) [inline babbage.nomad:111]: would have thought this would have less weight than AZ?

Comment (gedge) [reply babbage.nomad:111]: perhaps more 🤔

Comment (redhug1) [reply babbage.nomad:111]: Thats not how spread works, they are AND'd together .. such that they are equally spread by availability zone and then equally spread by unique host id's (or the other way around).

Comment (gedge) [reply babbage.nomad:111]: hmm, that rings a bell

I was (mistakenly) thinking: what's the point in `weight` then?
i.e. to say "across AZ" is more/less important than "across hosts"

but it's more "I want 70% of instances when attribute==`X`"

sounds legit

Comment (gedge) [reply babbage.nomad:111]: ah, no, looking at the spec, I think I was right, originally

Comment (gedge) [reply babbage.nomad:111]: `percentage` is what my last-but-one comment referred to

**Review by gedge — APPROVED** [review]

**Pull request merged**